index.js

```diff
@@ -0,0 +1,58 @@
+console.log("problem 1")
+const sumAll = (num, sum =0)=>{
+    if(num<=0) return sum
+    sum+=num
+    num--;
+    return sumAll(num, sum)
+}
+console.log(sumAll(6))
+
+
+console.log("problem 2")
+const multArr = (arr,i =0, product =1) =>{
+    if(i>=arr.length) return product;
+    product*=arr[i]
+    i++
+    return multArr(arr,i,product)
+}
+console.log(multArr([5,5,1,2]))
+
+
+console.log("problem 3")
+const concatArr = (arr, i=0, str ="") =>{
+    if(i>=arr.length) return str
+    str+=(arr[i]+ " ") 
+    i++;
+    return concatArr(arr, i, str)
```

Comment:
The output of your function would give you an extra space at the end of your string.